delta-app/src/main/java/io/cdap/delta/store/AssessmentServiceClient.java

```diff
@@ -96,37 +92,41 @@ public AssessmentServiceClient(ServiceDiscoverer context) {
    * Copied from "AbstractServiceRetryableMacroEvaluator"
    * Returns the full content of the error stream for the given {@link HttpURLConnection}.
    */
-  private <T> T validateAndRetrieveContent(@Nullable HttpURLConnection urlConn, Class<T> type) throws IOException {
+  private <T> T validateAndRetrieveContent(@Nullable HttpURLConnection urlConn, Type type) throws IOException {
     String serviceName = "State Store (AssessorService)";
     if (urlConn == null) {
       throw new RetryableException(serviceName + " service is not available");
     }
 
+    if (type == null) {
+      throw new NullPointerException(" Type parameter cannot be null");
+    }
+
     int responseCode = urlConn.getResponseCode();
     if (responseCode != HttpURLConnection.HTTP_OK) {
       throw new RetryableException("Failed to call " + serviceName + " service with status " + responseCode + ": " +
                                      getError(urlConn));
     }
 
     try (InputStream inputStream = urlConn.getInputStream()) {
-      String contentType = urlConn.getContentType();
-      switch (contentType) {
-        case MediaType.APPLICATION_JSON: {
-          return GSON.fromJson(new InputStreamReader(inputStream, StandardCharsets.UTF_8), type);
-        }
-
-        case MediaType.APPLICATION_OCTET_STREAM: {
-          return (T) ByteStreams.toByteArray(inputStream);
-        }
-
-        default:
-          return (T) CharStreams.toString(new InputStreamReader(inputStream, StandardCharsets.UTF_8));
-      }
+      return type == byte[].class ?  (T) retrieve(inputStream) :  retrieve(inputStream, type);
     } finally {
       urlConn.disconnect();
     }
   }
 
+  private byte[] retrieve(InputStream inputStream) throws IOException {
+    return ByteStreams.toByteArray(inputStream);
+  }
+
+  private <T> T retrieve(InputStream inputStream, Type type) throws IOException {
+    if (type == String.class) {
```

Comment:
don't need this, gson should work on String.class

Reply:
Actually I tried it out , it was failing. 

`GSON.fromJson(reader, type)`  

**case1** : reader's content is proper JSON , type is an Object --> Works 

**case2** : reader's content is proper JSON , type is an String.class --> **Fails**

	Error    `Expected a string but was BEGIN_OBJECT at line 1 column 2` 

**case3** : reader's content is a string , type is an String.class --> **Fails**

	Error : `com.google.gson.stream.MalformedJsonException`

	A workaround for case3 was using "JsonReader.setLenient(true)" , but it has it's own issues like it returns 1 word 
         , and have to iterate to get whole sentence.

Reply:
not sure what content you're trying to deserialize, but:

```
    String x = "xyz";
    String y = new Gson().fromJson(new StringReader(x), String.class);
```
this works fine for me. It's a bit of a moot point, you shouldn't be reading anything as a string anyway. It's only byte[] and OffsetAndSequence